dropshot/src/server.rs

```diff
@@ -543,18 +543,55 @@ impl<C: ServerContext> Service<&TlsConn> for ServerConnectionHandler<C> {
     }
 }
 
-/**
- * A running Dropshot HTTP server.
- */
-pub struct HttpServer<C: ServerContext> {
+/// A running Dropshot HTTP server.
+///
+/// The generic traits represent the following:
+/// - C: Caller-supplied server context
+/// - Closer: Identifies if the server is directly closeable.
+/// - Joiner: Identifies if the server is directly joinable.
```

Comment:
Tell me if I've got this right: the way I figure this working is that:

- Callers will get an `HttpServer<C, Closeable, Joinable>` without having specified `Closeable` or `Joinable` directly.  (They _will_ specify `C`.)
- The functions that get the `Closeable` or `Joinable` consume the `HttpServer` and return a new one with a different type (different type parameters for `Closer` or `Joiner`, respectively).

As a result, consumers mostly don't have to think about these type parameters.  Is that right?

The exception I can think of is that if you want to write a function that accepts an `HttpServer`, I expect you usually won't care whether about the `Closer` or `Joiner`, but your function will probably have to be generic over theses parameters.  That seems not ideal but probably a fine tradeoff.  This doesn't seem super common.

Reply:
That's right - though with this latest PR, the `Joinable` generic parameter has been removed.